v1/remote/error_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/google/go-containerregistry/v1/v1util"
+)
+
+func TestCheckErrorNil(t *testing.T) {
+	tests := []int{
+		http.StatusOK,
+		http.StatusAccepted,
+		http.StatusCreated,
+		http.StatusMovedPermanently,
+		http.StatusInternalServerError,
+	}
+
+	for _, code := range tests {
+		resp := &http.Response{StatusCode: code}
+
+		if err := checkError(resp); err != nil {
+			t.Errorf("checkError(%d) = %v", code, err)
+		}
+	}
+}
+
+func TestCheckErrorNotError(t *testing.T) {
+	tests := []struct {
+		code int
+		body string
+	}{{
+		code: http.StatusBadRequest,
+		body: "",
+	}, {
+		code: http.StatusUnauthorized,
+		body: "Not JSON",
+	}}
+
+	for _, test := range tests {
+		resp := &http.Response{
+			StatusCode: test.code,
+			Body:       v1util.NopReadCloser(bytes.NewBufferString(test.body)),
```

Comment:
`strings.NewReader` is shorter.

I forget what the benefit of `v1util.NopReadCloser` is over `ioutil.NopCloser`

Reply:
I bias towards `bytes.Buffer` because it is just so much more capable, but I can change if you feel strongly.

I only added it for naming consistency, where we also have `NopWriteCloser`.